libraries/libmesosphere/source/kern_k_memory_block_manager.cpp

```
@@ -347,43 +334,39 @@ namespace ams::kern {
             const size_t remaining_size = remaining_pages * PageSize;
-            KMemoryInfo cur_info = it->GetMemoryInfo();
 
             if ((it->GetAttribute() & mask) != attr) {
                 /* If we need to, create a new block before and insert it. */
-                if (cur_info.GetAddress() != GetInteger(cur_address)) {
+                if (it->GetAddress() != GetInteger(cur_address)) {
                     KMemoryBlock *new_block = allocator->Allocate();
 
                     it->Split(new_block, cur_address);
                     it = m_memory_block_tree.insert(*new_block);
                     it++;
```

cppcheck

[performance] Prefer prefix ++/-- operators for non-primitive types.

(postfixOperator)